### PR TITLE
Add values to control the Java heap size to Storm chart

### PR DESCRIPTION
**Pull request by ljubon**
used default values from https://github.com/apache/storm/blob/3f96c249cbc17ce062491bfbb39d484e241ab168/integration-test/config/storm.yaml#L40-L43

issue: https://github.com/G-Research/charts/issues/41

### CODE REVIEW

File: src/storm/templates/configmap.yaml [hunk 1]
```diff
@@ -21,6 +21,10 @@ data:
     storm.log4j2.conf.dir: /log4j2
     storm.local.dir: /data
     storm.log.dir: /logs
+    nimbus.childopts: {{ $.Values.nimbus.childopts }}
+    supervisor.childopts: {{ $.Values.supervisor.childopts }}
+    worker.childopts: {{ $.Values.worker.childopts }}
```
Comment (jgiannuzzi) [inline src/storm/templates/configmap.yaml:26]: I think we should also expose `worker.heap.memory.mb`, as it's a simpler one for defining the worker heap size

File: src/storm/values.yaml [hunk 2]
```diff
@@ -78,3 +81,5 @@ zookeeper:
   #  - "example.external.zookeeper3"
   # # external server port
   # port: 2181
+worker:
+  childopts: "-Xmx%HEAP-MEM%m -XX:+PrintGCDetails -Xloggc:artifacts/gc.log -XX:+PrintGCDateStamps -XX:+PrintGCTimeStamps -XX:+UseGCLogFileRotation -XX:NumberOfGCLogFiles=10 -XX:GCLogFileSize=1M -XX:+HeapDumpOnOutOfMemoryError -XX:HeapDumpPath=artifacts/heapdump -ea"
```
Comment (jgiannuzzi) [inline src/storm/values.yaml:85]: any `worker` options belong to `supervisor`, as it's the one responsible for spawning workers

Comment (ljubon) [reply src/storm/values.yaml]: should I remove `worker.childopts` in that case? 🤔

File: src/storm/templates/configmap.yaml (hunk 1 above)
Comment (ljubon) [reply src/storm/templates/configmap.yaml]: added 💪

File: src/storm/values.yaml (hunk 2 above)
Comment (jgiannuzzi) [reply src/storm/values.yaml]: I think we could remove it for now, yes.

To go back to my initial comment, please be sure to rename all the `worker.*` values to `supervisor.worker.*` in `values.yaml`!

Comment (ljubon) [reply src/storm/values.yaml]: done 🚀

File: src/storm/values.yaml [hunk 3]
```diff
@@ -30,10 +31,15 @@ supervisor:
...
     port: 6700
+  childopts: "-Xmx256m -ea"
   resources: {}
   nodeSelector: {}
   tolerations: []
   affinity: {}
+  worker:
+    heap:
+      memory:
+        mb: 768
```
Comment (jgiannuzzi) [inline src/storm/values.yaml:42]: question: what about renaming this to `heap_memory_mb`?

Comment (ljubon) [reply src/storm/values.yaml]: cleaner, like it 👍 
at first, i wanted to re-use the same names as original ⚡

**Review by jgiannuzzi — APPROVED** [review]
LGTM 👍

Thanks @ljubon!

**Pull request merged**